code/modules/cooking/plates.dm

```diff
@@ -0,0 +1,186 @@
+        return
+    if (istype(I, /obj/item/reagent_containers/food/snacks) || istype(I, /obj/item/trash))
+        to_chat(user, SPAN_WARNING("\The [name] already has something on it!"))
+        return
+    if (istype(I, /obj/item/material/kitchen/utensil) && istype(holding, /obj/item/reagent_containers/food/snacks))
+        var/obj/item/temp_hold = holding.attackby(I, user)
+        if(temp_hold != holding)
+            holding = temp_hold
+            user.prepare_for_slotmove(temp_hold)
+            temp_hold.forceMove(src)
+            if(!grease) //Grease lightning, go grease lightning!
+                grease = TRUE
+        update_icon()
+        return
+    if (istype(I, /obj/item/material/kitchen/utensil) && istype(holding, /obj/item/trash))
+        to_chat(user, SPAN_WARNING("You're not sure you should try to eat \the [holding.name]."))
+    if (istype(I, /obj/item/material/kitchen/utensil))
+        to_chat(user, SPAN_WARNING("There isn't any food on \the [name]."))
+        return
+    return ..()
+
+/obj/item/plate/on_rag_wipe(obj/item/reagent_containers/glass/rag/R)
+    . = ..()
+    if(grease)
+        grease = FALSE
+    return
+
+/obj/item/plate/attack_self(mob/user)
+    if(!user.get_inactive_hand()) //Free hand
+        var/obj/item/reagent_containers/food/snacks/F = holding
+        user.put_in_hands(F)
+        holding = null
+        update_icon()
+        to_chat(user, SPAN_NOTICE("You take \the [F.name] from \the [name]"))
+        return
+    return ..()
+
+/obj/item/plate/attack(mob/living/M, mob/living/user, target_zone)
+    if(istype(holding, /obj/item/reagent_containers/food/snacks))
+        var/obj/item/reagent_containers/food/snacks/S = holding 
+        var/obj/item/temp_hold = S.standard_feed_mob(user, M)
+        if(temp_hold != holding)
+            holding = temp_hold
+            user.prepare_for_slotmove(temp_hold)
+            temp_hold.forceMove(src)
+            if(!grease) //Grease lightning, go grease lightning!
+                grease = TRUE
+        update_icon()
+    else if(grease && !holding && (M == user))
+        user.visible_message(SPAN_NOTICE("[user] starts to lick \the [src] clean."), SPAN_NOTICE("You start to lick \the [src] clean."))
+        if(do_after(user, 5))
+            grease = FALSE
+            user.visible_message(SPAN_NOTICE("[user] licks everything off \the [src]."), SPAN_NOTICE("You lick everything off \the [src]."))
+    else
+        return ..()
+    update_icon()
+    return
+     
+/obj/item/plate/update_icon()
+    var/list/O = list()
+    if(grease)
+        icon_state = "[initial(icon_state)]_mess"
+    else
+        icon_state = initial(icon_state)
+    if(holding)
+        holding.update_icon() //Just to be safe
+        LAZYADD(O, image(icon=holding.icon, icon_state=holding.icon_state))
+    set_overlays(O)
+    return ..()
+
+//Is this technically a plate? No. Was it created to do a similar function? Yes
+/obj/item/reagent_containers/bowl
```

Comment:
This is a bit weird - I see that they share some code. Why isn't this a subtype of the other, or viceversa?